integration_tests/src/main/python/regexp_test.py

```diff
@@ -808,7 +808,7 @@ def test_regexp_extract_all_idx_negative():
 
 @allow_non_gpu('ProjectExec', 'RegExpExtractAll')
 def test_regexp_extract_all_idx_out_of_bounds():
-    gen = mk_str_gen('[abcd]{0,3}')
+    gen = StringGen('.{0,10}')
```

Comment:
Sorry for the late reply.

The test is failed after this PR because the error message "Regex group count is 2, but the specified group index is 3" will only raised when data matches the pattern. 

In the previous code, what triggers this error message is actually the `'.{0,10}'` as a special case in `mk_str_gen`. So the test was PASSED because of good luck.

Now I change the pattern to `'[a-d]{1,2}.{0,1}[0-9]{1,2}'` to make sure they can match the pattern below.